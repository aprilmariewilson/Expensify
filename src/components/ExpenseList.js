import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

//this is the component with the information from the store
export const ExpenseList = (props) => (
    <div>
    {
        props.expenses.length === 0 ? (
            <p>No expenses</p>
        ) : (
            props.expenses.map((expense) => {
                return <ExpenseListItem key= {expense.id} {...expense}/>
            })
        )         
    }

        
    </div>
);
//create new higher order coponents using the connect function from reaxt/redux 
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)

    };
};

// call connect and define what we want from the store and define component we want to create the connected version of
export default connect(mapStateToProps)(ExpenseList);