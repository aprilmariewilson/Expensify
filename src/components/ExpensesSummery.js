import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/selectExpenseTotal';
import numeral from 'numeral';

//show how many items and a running total for the items
export const ExpensesSummery = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format(' $0,0.00');

    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}. </h1>
        </div>
    );
};
const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummery);