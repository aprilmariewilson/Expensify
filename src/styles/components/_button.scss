.login__button {
    background: $blue;
    font: url(/fonts/ProductSans-Regular.ttf);
    border: $blue;
    font-size: 2.2rem;
    color: white;
    font-weight: 300; 
    padding: $m-size $s-size;
} 
.button {
    background: $blue;
    border: none;
    display: inline-block;
    font-size: $lg-font;
    color: white;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}
.button__logout {
    background: $dark-blue;
    color: white;
    font-weight: 400;
}
.button__remove {
    background: #888;
}