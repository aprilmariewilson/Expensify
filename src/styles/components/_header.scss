.header {
    background: $dark-blue;
}
.header__title {
color: white;
text-decoration: none;
h1 {
    font-size: 4.8rem;
    line-height: 1;
  }
}
.header__content {
    align-items: center;
    display: flex;
    justify-content: space-between;
}