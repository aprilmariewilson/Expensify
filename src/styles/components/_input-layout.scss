.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $m-size;
    @media (min-width: 45rem) {
      flex-direction: row;
      margin-bottom: $l-size;
    }
  }
  
  .input-group__item {
    margin-bottom: $s-size;
    @media (min-width: 45rem) {
      margin: 0 $s-size 0 0;
    }
  }
  