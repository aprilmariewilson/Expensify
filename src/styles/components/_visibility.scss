.show-for-mobile {
    @media (min-width: 45rem) {
      display: none;
    }
  }
  
  .show-for-desktop {
    @media (max-width: 44.99rem) {
      display: none;
    }
  }