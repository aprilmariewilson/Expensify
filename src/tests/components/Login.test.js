import React from 'react';
import { shallow } from 'enzyme';
import { LogInPage } from '../../components/Login';


test('should render the LogInPage correctly', () => {
    const wrapper = shallow(<LogInPage startLogIn={() => { }}/>);
    expect(wrapper).toMatchSnapshot();
});
test('should call startLogIn on button click', () => {
    const startLogin=jest.fn();
    const wrapper = shallow(<LogInPage />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled
});