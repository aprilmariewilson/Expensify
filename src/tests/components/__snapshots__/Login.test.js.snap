// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the LogInPage correctly 1`] = `
<div>
  <h2>
    Please Log In
  </h2>
  <button>
    Log In
  </button>
</div>
`;
